app/jobs/request-payment-status.js

```diff
@@ -0,0 +1,157 @@
+import { MessageReceiver } from 'ffc-messaging'
+import { config } from '../config/index.js'
+import { sendMessage } from '../messaging/send-message.js'
+import { sendPaymentDataRequest } from '../messaging/send-payment-data-request.js'
+import {
+  getPendingPayments,
+  incrementPaymentCheckCount,
+  updatePaymentStatusByClaimRef
+} from '../repositories/payment-repository.js'
+import { createBlobServiceClient } from '../storage.js'
+import { v4 as uuid } from 'uuid'
+import { Status } from '../constants/constants.js'
+
+const {
+  messageQueueConfig: {
+    moveClaimToPaidMsgType,
+    applicationRequestQueue,
+    paymentDataRequestResponseQueue
+  },
+  storageConfig: {
+    paymentDataHubConnectionString,
+    paymentDataHubDataRequestsContainer
+  }
+} = config
+
+const createPaymentDataRequest = (frn) => ({
+  category: 'frn',
+  value: frn
+})
+
+const processPaidClaim = async (claimReference, logger) => {
+  const [, updatedRows] = await updatePaymentStatusByClaimRef(claimReference, Status.PAID)
+
+  if (updatedRows?.length === 1) {
+    await sendMessage(
+      {
+        claimRef: claimReference,
+        sbi: updatedRows[0].dataValues.sbi
+      },
+      moveClaimToPaidMsgType,
+      applicationRequestQueue,
+      { sessionId: uuid() }
+    )
+  } else {
+    logger.error('Payment not found to update paid status')
+  }
+}
+
+const processPaymentDataBlob = async (paymentDataBlob, claimReferences, logger) => {
+  if (!claimReferences.has(paymentDataBlob.agreementNumber)) { return }
+
+  const { agreementNumber: claimReference, status } = paymentDataBlob
+  logger.setBindings({ claimReference, status })
+
+  if (status.state === Status.PAID) {
+    await processPaidClaim(claimReference, logger)
+  } else {
+    await incrementPaymentCheckCount(claimReference)
+  }
+}
+
+const processDataRequestResponse = async ({ logger, blobServiceClient, claimReferences, blobUri }) => {
+  const blob = await blobServiceClient.getBlob(
+    logger,
+    blobUri,
+    paymentDataHubDataRequestsContainer
+  )
+
+  for (const paymentDataBlob of blob.data) {
+    await processPaymentDataBlob(paymentDataBlob, claimReferences, logger)
+  }
+}
+
+const createReceiver = async (messageId) => {
+  const receiver = new MessageReceiver(paymentDataRequestResponseQueue)
+  await receiver.acceptSession(messageId)
+  return receiver
+}
+
+const processFrnRequest = async (frn, logger, claimReferences, blobServiceClient) => {
+  const requestMessageId = uuid()
+  const sessionId = uuid()
+  const requestMessage = createPaymentDataRequest(frn)
+
+  logger.setBindings({ frn, messageId: requestMessageId })
+
+  let receiver
+  let responseMessage
+  let blobUri
```

Comment:
let receiver, responseMessage, blobUri
```suggestion
  let receiver
  let responseMessage
  let blobUri
```